### PR TITLE
api: ext-proc processing mode

### PR DESCRIPTION
**Pull request by guydc**
**What this PR does / why we need it**:
Allows users to define how request and response header/body are processed by exteral processing service. 

By default, nothing is sent to the external processor. If an empty request/response struct is defined, headers of request/response are sent. If the body attribute is configured, the body is processed according to the defined processing mode (Stream, Buffer, BufferPartial).

**Which issue(s) this PR fixes**:
xref: #3170

### CODE REVIEW

File: test/cel-validation/envoyextensionpolicy_test.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ package celvalidation
 import (
 	"context"
 	"fmt"
+	"k8s.io/utils/ptr"
```
Comment (shawnh2) [inline test/cel-validation/envoyextensionpolicy_test.go:14]: sort the imports please

Comment (zirain) [reply test/cel-validation/envoyextensionpolicy_test.go:14]: please wait https://github.com/envoyproxy/gateway/pull/3177

**Comment by zirain** [comment]
@guydc sorry, linter works as expected now, can you fix it?

**Comment by zirain** [comment]
@guydc sorry, linter works as expected now, can you fix it?

**Review by zirain — COMMENTED** [review]
LGTM

**Review by shawnh2 — APPROVED** [review]
lgtm

File: api/v1alpha1/ext_proc_types.go [hunk 2]
```diff
@@ -9,6 +9,39 @@ import (
 	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
 )
 
+// +kubebuilder:validation:Enum=Streamed;Buffered;BufferedPartial
+type ExtProcBodyProcessingMode string
+
+const (
+	StreamedExtProcBodyProcessingMode          ExtProcBodyProcessingMode = "Streamed"
```
Comment (zhaohuabing) [inline api/v1alpha1/ext_proc_types.go:16]: Can we add comments for each mode so they can be generated in API docs?

**Comment by zhaohuabing** [comment]
Minor/non-blocking comment on naming:

`ProcessingMode` is not a very clear name, but I can't think of a better name either.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Comment by arkodg** [comment]
> Minor/non-blocking comment on naming:
> 
> `ProcessingMode` is not a very clear name, but I can't think of a better name either.

yeah this API field is doing two things
* what to send (body, header)
* when to send it (stream, buffered)

can't think of a better name, so lets go ahead with the envoy specific name, since this feature is completely tied to envoy

**Review by zhaohuabing — APPROVED** [review]
LGTM, thanks.

**Pull request merged**